proc/traverse/over.go

```diff
@@ -0,0 +1,100 @@
+package traverse
+
+import (
+	"github.com/brimdata/zed"
+	"github.com/brimdata/zed/expr"
+	"github.com/brimdata/zed/proc"
+	"github.com/brimdata/zed/zbuf"
+	"github.com/brimdata/zed/zcode"
+)
+
+type Over struct {
+	exprs  []expr.Evaluator
+	parent proc.Interface
+	batch  zbuf.Batch
+	vals   []zed.Value
+	eof    bool
+}
+
+func NewOver(parent proc.Interface, exprs []expr.Evaluator) *Over {
+	return &Over{
+		exprs:  exprs,
+		parent: parent,
+	}
+}
+
+func (o *Over) Pull() (zbuf.Batch, error) {
+	if len(o.vals) == 0 {
+		batch, err := o.parent.Pull()
+		if batch == nil || err != nil {
+			return batch, err
+		}
+		o.eof = false
+		o.batch = batch
+		o.vals = batch.Values()
+	}
+	if o.eof {
+		o.eof = false
+		return nil, nil
+	}
+	o.eof = true
+	out, err := o.over(&o.vals[0])
+	o.vals = o.vals[1:]
+	if len(o.vals) == 0 {
+		o.batch.Unref()
+	}
```

Comment:
This is unsafe since `out.Bytes` might refer to `o.vals[0].Bytes`.
```suggestion
```

Reply:
There's a copy for this reason.  Better to send it to GC?

We should discuss this protocol.  I think a much better buffering model that isn't the full blown "mbuf" proposal from long ago shouldn't be too hard.

Reply:
I missed the copy, so this is indeed fine.